### PR TITLE
feat(app): add static dynamic-plugins-info

### PR DESCRIPTION
**Pull request by gashcrumb**
## Description

This change adds a means to statically add and configure a frontend plugin that needs to contribute UI elements to any of the areas of the UI managed by the dynamic frontend implementation.  This commit also adds the dynamic-plugins-info frontend plugin and provides a static base configuration for it.  Finally, this change cleans up a couple of development time warnings that get logged at the javascript console.

## Which issue(s) does this PR fix

- Fixes https://issues.redhat.com/browse/RHIDP-1650

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

## How to test changes / Special notes to the reviewer

In the app, look at Administration -> Plugins; there should not be a `@janus-idp/backstage-plugin-dynamic-plugins-info` entry present in the table, for example:

image

### CODE REVIEW

File: showcase-docs/dynamic-plugins.md [hunk 1]
```diff
@@ -619,6 +619,8 @@ The following mount points are available:
 
 | Mount point                  | Description                         | Visible even when no plugins are enabled                       |
 | ---------------------------- | ----------------------------------- | -------------------------------------------------------------- |
+| `admin.page.plugins`         | Administration plugins page         | NO                                                             |
+| `admin.page.rbac`            | Administration RBAC page            | NO                                                             |
```
Comment (invincibleJai) [inline showcase-docs/dynamic-plugins.md:623]: Will this impact rbac as well?

**Comment by invincibleJai** [comment]
thanks @gashcrumb , i tried the PR locally and i could see it

image


is there anything I could be missing?

File: plugins/dynamic-plugins-info/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Setting up the development environment for Dynamic Plugins Info plugin
+
+In [Backstage plugin terminology](https://backstage.io/docs/local-dev/cli-build-system#package-roles), the Dynamic Plugins Info plugin is a front-end plugin. You can start a live development session from the repository root using the following command:
+
+```console
+yarn workspace @janus-idp/backstage-plugin-dynamic-plugins-info run start
```
Comment (invincibleJai) [inline plugins/dynamic-plugins-info/CONTRIBUTING.md:6]: @janus-idp/backstage-plugin-dynamic-plugins-info might not be needed right or if this file needs an update too?

File: plugins/dynamic-plugins-info/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+## @janus-idp/backstage-plugin-dynamic-plugins-info [1.0.2](https://github.com/janus-idp/backstage-plugins/compare/@janus-idp/backstage-plugin-dynamic-plugins-info@1.0.1...@janus-idp/backstage-plugin-dynamic-plugins-info@1.0.2) (2024-02-21)
```
Comment (invincibleJai) [inline plugins/dynamic-plugins-info/CHANGELOG.md:1]: do we need to push this file as well?

**Comment by invincibleJai** [comment]
Thanks @gashcrumb , have verified the changes work as expected.

@gashcrumb so should we remove the plugin https://github.com/janus-idp/backstage-plugins/tree/main/plugins/dynamic-plugins-info from plugins repo as well?

File: plugins/dynamic-plugins-info/CONTRIBUTING.md (hunk 2 above)
Comment (gashcrumb) [reply plugins/dynamic-plugins-info/CONTRIBUTING.md:6]: Yep, I think I should probably clean up all of this for this PR.

File: plugins/dynamic-plugins-info/CHANGELOG.md (hunk 3 above)
Comment (gashcrumb) [reply plugins/dynamic-plugins-info/CHANGELOG.md:1]: No, I'll take this out, thanks!

**Comment by gashcrumb** [comment]
> Thanks @gashcrumb , have verified the changes work as expected.
> 
> @gashcrumb so should we remove the plugin [janus-idp/backstage-plugins@`main`/plugins/dynamic-plugins-info](https://github.com/janus-idp/backstage-plugins/tree/main/plugins/dynamic-plugins-info?rgh-link-date=2024-04-01T11%3A09%3A24Z) from plugins repo as well?

Yep, eventually.  I figure assuming we go in the direction this PR suggests I'll create an issue later on to remove that copy, or if we start moving plugins upstream in the meantime from backstage-plugins, then that copy of the plugin can be excluded.

**Comment by invincibleJai** [comment]
@gashcrumb once you address open comments LMK, will add lgtm

**Review by davidfestal — APPROVED** [review]

**Comment by gashcrumb** [comment]
Lemme rebase this once the e2e tests are fixed, I'd like to see them pass with this change in place.

**Comment by ciiay** [comment]
I have manually removed the dynamic-plugins-info package under `dynamic-plugins-root/` directory and removed dynamic-plugins-info configuration from `app-config.local.yaml` and tested this out. Works perfectly.
pr_1151_review

**Pull request merged**